### ISSUE TITLE
Run Configurations are not loaded at IDE startup

### ISSUE THREAD

**Issue by dschniedertuens-tc** [flutter/flutter-intellij#7139]
For several weeks now Android Studio doesn't load the run configurations for my flutter project that are stored as files when I start the IDE.

A workaround is to select "File" / "Repair IDE". Most of the time it restores the expected state after recovery step 2 "Rescan Project Indexes". But it will be lost again after I close the IDE and start it again.

I already filed an issue at the Android Studio issue tracker, but it was closed immediately refering to this flutter-intellij tracker.

Please let me know if you need any further information and thank you in advance for taking your time.

## Steps to Reproduce

1. Create some run configurations like these:
![RunConfigurations](https://github.com/flutter/flutter-intellij/assets/41641370/01df46c0-b03f-4024-bc59-e04e739ccb64)

2. Close Android Studio and start it again. The run configurations are missing and in addition to that the IDE generates an unwanted default configuration "main.dart":
![Empty_RunConfigurations](https://github.com/flutter/flutter-intellij/assets/41641370/c997c696-5ba0-415b-97fd-5aaaf352d91e)

## Version info

$ flutter doctor -v
[√] Flutter (Channel stable, 3.16.4, on Microsoft Windows [Version 10.0.19045.3803], locale de-DE)
    • Flutter version 3.16.4 on channel stable at C:\ProgramData\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2e9cb0aa71 (3 days ago), 2023-12-11 14:35:13 -0700
    • Engine revision 54a7145303
    • Dart version 3.2.3
    • DevTools version 2.28.4

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at C:\ProgramData\Android\Sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-b2043.56-10550314)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • CHROME_EXECUTABLE = C:\Projekte Zusatzdaten\Android\google-chrome-unsafe.bat

[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.8.3)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.8.34330.188
    • Windows 10 SDK version 10.0.20348.0

[√] Android Studio (version 2023.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-b2043.56-10550314)

[√] VS Code (version 1.84.2)
    • VS Code at C:\Users\Daniel\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.76.0

[√] Connected device (4 available)
    • sdk gphone64 x86 64 (mobile) • emulator-5554 • android-x64    • Android 14 (API 34) (emulator)
    • Windows (desktop)            • windows       • windows-x64    • Microsoft Windows [Version 10.0.19045.3803]
    • Chrome (web)                 • chrome        • web-javascript • Google Chrome 120.0.6099.71
    • Edge (web)                   • edge          • web-javascript • Microsoft Edge 120.0.2210.61

[√] Network resources
    • All expected network resources are available.

• No issues found!

**Comment by TDuffinNTU** [comment]
I think this is because Android studio doesn't seem to support projects with nested app folders? If you opened one of those app folders within android studio (file > open > ...) then you _should_ be able to run them. It's really annoying because it messes with our folder structure too ...

**Comment by dschniedertuens** [comment]
Yes, that’s what I suspect as well. In the meantime, I’ve tweaked the directory structure a bit and am using Dart’s new workspace feature. Before, I had individual project modules, which I think aren’t even supported anymore.

Now I open it as a single project from the project root directory, which contains multiple apps and packages. As a result, the run configurations have also moved into a directory relative to the project root, and there were no more issues with loading. So I couldn’t even tell you now if the original problem still exists.

Maybe this approach could work for your project, too.

**Comment by TDuffinNTU** [comment]
I was about to reply myself, but it appears that there's a feature in android studio called "project modules" that lets you set up each folder as its own sub-project (eg. run configs). I'm testing it as we speak but it seems to work for my case.

`file > project structure > + > import module`

**Comment by dschniedertuens** [comment]
From my experiences this feature messes even more with project settings. And I think I read somewhere that this feature is no longer supported.
